enterprise/server/ociregistry/ociregistry.go

```diff
@@ -190,20 +237,197 @@ func (r *registry) handleBlobsOrManifestsRequest(ctx context.Context, w http.Res
 	}
 	defer upresp.Body.Close()
 
-	if upresp.Header.Get(headerWWWAuthenticate) != "" {
-		w.Header().Add(headerWWWAuthenticate, upresp.Header.Get(headerWWWAuthenticate))
+	for _, header := range []string{headerContentLength, headerContentType, headerDockerContentDigest, headerWWWAuthenticate} {
+		if upresp.Header.Get(header) != "" {
+			w.Header().Add(header, upresp.Header.Get(header))
+		}
 	}
-	w.Header().Add(headerContentType, upresp.Header.Get(headerContentType))
-	w.Header().Add(headerDockerContentDigest, upresp.Header.Get(headerDockerContentDigest))
-	w.Header().Add(headerContentLength, upresp.Header.Get(headerContentLength))
 	w.WriteHeader(upresp.StatusCode)
-	_, err = io.Copy(w, upresp.Body)
+
+	hasContentLength := upresp.Header.Get(headerContentLength) != ""
+	var contentLength int64
+	if hasContentLength {
+		if positiveWholeNumberRegexp.MatchString(upresp.Header.Get(headerContentLength)) {
```

Comment:
is this regexp necessary? could you check the value returned by ParseInt instead?